packages/leafygreen-provider/src/LeafyGreenProvider.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { createContext, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useEventListener } from '@leafygreen-ui/hooks';
+
+interface UsingKeyboardState {
+  usingKeyboard?: boolean;
+  setUsingKeyboard?: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const UsingKeyboardContext = createContext<UsingKeyboardState>({
+  usingKeyboard: undefined,
+  setUsingKeyboard: undefined,
+});
+
+export const NavigationKeyCodes = {
+  tab: 9,
+  leftArrow: 37,
+  upArrow: 38,
+  rightArrow: 39,
+  downArrow: 40,
+} as const;
+
+type NavigationKeyCodes = typeof NavigationKeyCodes[keyof typeof NavigationKeyCodes];
+
+interface LeafyGreenProviderProps {
+  children: React.ReactNode;
+  initialState?: {
+    usingKeyboard?: boolean;
+  };
+}
+
+const LeafyGreenProvider = ({
```

Comment:
Ideally this is a very simple component. What you have here is the former `KeyboardContextProvider`. To prepare for future context's we should still have `KeyboardContextProvider` and have this component merely use that provider. Such that it becomes trivial to add or remove contexts in the future. 

I'd even argue to make UsingKeyboard related code its own file.